diff --git a/node_modules/@dongminyu/react-native-step-counter/android/build.gradle b/node_modules/@dongminyu/react-native-step-counter/android/build.gradle
index e4ee6d5..611d8d7 100644
--- a/node_modules/@dongminyu/react-native-step-counter/android/build.gradle
+++ b/node_modules/@dongminyu/react-native-step-counter/android/build.gradle
@@ -62,7 +62,7 @@ android {
   compileSdkVersion getExtOrIntegerDefault("compileSdkVersion")
 
   defaultConfig {
-    minSdkVersion getExtOrIntegerDefault("minSdkVersion")
+    minSdkVersion 24
     targetSdkVersion getExtOrIntegerDefault("targetSdkVersion")
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
   }
diff --git a/node_modules/@dongminyu/react-native-step-counter/android/gradle.properties b/node_modules/@dongminyu/react-native-step-counter/android/gradle.properties
index a805b86..1830a9a 100644
--- a/node_modules/@dongminyu/react-native-step-counter/android/gradle.properties
+++ b/node_modules/@dongminyu/react-native-step-counter/android/gradle.properties
@@ -1,5 +1,5 @@
-StepCounter_compileSdkVersion=31
-StepCounter_kotlinVersion=1.7.0
-StepCounter_targetSdkVersion=31
-StepCounter_minSdkVersion=21
-StepCounter_ndkversion=21.4.7075529
+StepCounter_compileSdkVersion=34
+StepCounter_kotlinVersion=1.9.23
+StepCounter_targetSdkVersion=34
+StepCounter_minSdkVersion=24
+StepCounter_ndkversion=26.1.10909125
\ No newline at end of file
diff --git a/node_modules/@dongminyu/react-native-step-counter/android/src/main/java/com/stepcounter/StepCounterModule.kt b/node_modules/@dongminyu/react-native-step-counter/android/src/main/java/com/stepcounter/StepCounterModule.kt
index 7e39a7b..d6ab0f8 100644
--- a/node_modules/@dongminyu/react-native-step-counter/android/src/main/java/com/stepcounter/StepCounterModule.kt
+++ b/node_modules/@dongminyu/react-native-step-counter/android/src/main/java/com/stepcounter/StepCounterModule.kt
@@ -35,7 +35,7 @@ class StepCounterModule internal constructor(context: ReactApplicationContext) :
     }
 
     private val appContext: ReactApplicationContext = context
-    private val lateinit sensorManager: SensorManager
+    private var sensorManager: SensorManager
     private val stepsOK: Boolean
         get() = checkSelfPermission(appContext, STEP_COUNTER) == PERMISSION_GRANTED
     private val accelOK: Boolean
